src/modules/servicenode.js

```diff
@@ -10,7 +10,20 @@ class ServiceNode {
         this.handler = message => {
             return self.handleMessage(message);
         };
+        this.pinner = async multihash => {
+            try {
+                this.chluIpfs.pinning.pin(multihash);
+            } catch (error) {
+                this.chluIpfs.logger.error('Pinning failed due to Error: ' + error.message);
+            }
+        };
+        // Handle Chlu network messages
         this.chluIpfs.events.on('message', this.handler);
+        // Pin public keys
```

Comment:
As I understand these events are only emitted when we want to pin the data. Should we instead use an async function that is called instead of going via an event? What advantage do we get from using events as an indirection here?

Also, the event name is again a problem. Maybe they should be something like `serviceNode/pinVendorPubKey`?

Reply:
the event is actually emitted when ChluIPFS discovers a public key in a __valid__ review record. The point of putting the pin logic here is to separate the service node from the rest of the library, I thought in the future we will need the service node to expose an HTTP API and other stuff and it will probably get its own repository so we might as well start separating concerns now.

Let me know what you think